tests/Report/JunitReporterTest.php

```diff
@@ -0,0 +1,265 @@
+    {
+        $expectedXml = <<<'XML'
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites>
+  <testsuite name="PHP CS Fixer report" tests="1" assertions="1" failures="1" errors="0">
+    <testcase name="someFile" class="someFile" file="someFile.php" assertions="1">
+      <failure type="csc">Wrong code style</failure>
+      <system-err><![CDATA[Suggested fix:
+--------------
+
+this text is a diff ;)]]></system-err>
+    </testcase>
+  </testsuite>
+</testsuites>
+XML;
+
+        $actualXml = $this->reporter->generate(
+            ReportSummary::create()
+                ->setChanged(
+                    array(
+                        'someFile.php' => array(
+                            'appliedFixers' => array('some_fixer_name_here'),
+                            'diff' => 'this text is a diff ;)',
+                        ),
+                    )
+                )
+        );
+
+        $this->assertJunitXmlSchema($actualXml);
+        $this->assertXmlStringEqualsXmlString($expectedXml, $actualXml);
+    }
+
+    public function testGenerateWithAppliedFixers()
+    {
+        $expectedXml = <<<'XML'
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites>
+  <testsuite name="PHP CS Fixer report" tests="1" assertions="2" failures="2" errors="0">
+    <testcase name="someFile" class="someFile" file="someFile.php" assertions="2">
+      <failure type="some_fixer_name_here_1">Some fixer name here 1</failure>
+      <failure type="some_fixer_name_here_2">Some fixer name here 2</failure>
+    </testcase>
+  </testsuite>
+</testsuites>
+XML;
+
+        $actualXml = $this->reporter->generate(
+            ReportSummary::create()
+                ->setAddAppliedFixers(true)
+                ->setChanged(
+                    array(
+                        'someFile.php' => array(
+                            'appliedFixers' => array('some_fixer_name_here_1', 'some_fixer_name_here_2'),
+                        ),
+                    )
+                )
+        );
+
+        $this->assertJunitXmlSchema($actualXml);
+        $this->assertXmlStringEqualsXmlString($expectedXml, $actualXml);
+    }
+
+    public function testGenerateWithTimeAndMemory()
+    {
+        $expectedXml = <<<'XML'
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites>
+  <testsuite name="PHP CS Fixer report" tests="1" assertions="1" failures="1" errors="0" time="1.234">
+    <testcase name="someFile" class="someFile" file="someFile.php" assertions="1">
+      <failure type="csc">Wrong code style</failure>
+    </testcase>
+  </testsuite>
+</testsuites>
+XML;
+
+        $actualXml = $this->reporter->generate(
+            ReportSummary::create()
+                ->setChanged(
+                    array(
+                        'someFile.php' => array(
+                            'appliedFixers' => array('some_fixer_name_here'),
+                        ),
+                    )
+                )
+                ->setMemory(2.5 * 1024 * 1024)
```

Comment:
sorry, I see memory, but think about time =))

May be should keep this to assert that setting memory does not affect xml because schema does not support it (only time) ?